.github/workflows/ci.yml

```diff
@@ -8,9 +8,21 @@ on:
 jobs:
   test:
     runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        stack-yaml:
+          - stack.yaml
+          - stack-lts-18.0.yaml
+          - stack-lts-18.25.yaml
+      fail-fast: false
+
     steps:
       - uses: actions/checkout@v2
-      - uses: freckle/stack-cache-action@v1.0.1
-      - uses: freckle/stack-action@main
+      - uses: freckle/stack-cache-action@v2
+        with:
+          stack-yaml: ${{ matrix.stack-yaml }}
+      - uses: freckle/stack-action@v3.1.4
         with:
           weeder: false
```

Comment:
You can remove this. The new version of the Action doesn't run weeder. You'll probably want to add an `hlint` Job too.